woocommerce-gateway-stripe.php

```diff
@@ -308,6 +309,26 @@ public function filter_gateway_order_admin( $sections ) {
 				return $sections;
 			}
 
+			/**
+			 * Provide default values for missing settings on initial gateway settings save.
+			 *
+			 * @since 4.5.4
+			 * @version 4.5.4
+			 *
+			 * @param array $settings New settings to save
+			 * @param array|bool $old_settings Existing settings, if any.
+			 * @return array New value but with defaults initially filled in for missing settings.
+			 */
+			public function gateway_settings_update( $settings, $old_settings ) {
+				if ( false === $old_settings ) {
+					$gateway  = new WC_Gateway_Stripe();
+					$fields   = $gateway->get_form_fields();
+					$defaults = array_merge( array_fill_keys( array_keys( $fields ), '' ), wp_list_pluck( $fields, 'default' ) );
```

Comment:
The implementation here is borrowed from [this core code](https://github.com/woocommerce/woocommerce/blob/4240f9924ad29d69920e0c40ed86598ec157901c/includes/abstracts/abstract-wc-settings-api.php#L266-L267).

Note the similarity to [this function](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/95d5eead79f47c485c5014fcb7e8a0a763b6aac9/includes/connect/class-wc-stripe-connect.php#L169-L183) (recently migrated from WooCommerce Services) – one or both of these uses could later be refactored to use the same code.

Reply:
Yeah, the recent WCS additions are nearly identical. We can move forward with this but lets open an issue to resolve the tech debt.